diff --git a/PKGBUILD b/PKGBUILD
index ee77260..77693db 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,6 +1,6 @@
 # Maintainer: Jan Alexander Steffens (heftig) <heftig@archlinux.org>
 
-pkgbase=linux
+pkgbase=linux-tlsm
 pkgver=6.18.9.arch1
 pkgrel=2
 pkgdesc='Linux'
@@ -35,9 +35,12 @@ options=(
 _srcname=linux-${pkgver%.*}
 _srctag=v${pkgver%.*}-${pkgver##*.}
 source=(
-  https://cdn.kernel.org/pub/linux/kernel/v${pkgver%%.*}.x/${_srcname}.tar.{xz,sign}
-  $url/releases/download/$_srctag/linux-$_srctag.patch.zst{,.sig}
+  https://cdn.kernel.org/pub/linux/kernel/v${pkgver%%.*}.x/${_srcname}.tar.xz
+  $url/releases/download/$_srctag/linux-$_srctag.patch.zst
   config  # the main kernel config file
+  tlsm.tar.gz
+  Kconfig.diff
+  Makefile.diff
 )
 validpgpkeys=(
   ABAF11C65A2970B130ABE3C479BE3E4300411886  # Linus Torvalds
@@ -46,15 +49,17 @@ validpgpkeys=(
 )
 # https://www.kernel.org/pub/linux/kernel/v6.x/sha256sums.asc
 sha256sums=('030115ff8fb4cb536d8449dc40ebc3e314e86ba1b316a6ae21091a11cc930578'
-            'SKIP'
             '4815407239a6df15f8e0362ff652f9faf2e558fd774b08645e80ca664128e390'
             'SKIP'
-            '9fed188f89847418aaf6416b64457a30bee34dcd0fa42a84dbd0f4dfca063402')
+            'SKIP'
+            'SKIP'
+            'SKIP')
 b2sums=('9aed902e41583597cb7595efe77504630a621993d20f89365a93cf2ea4d9790a6361d93cbb7fd7603881a4f82b76394b7e12fb4e4a88c9fedb2d63d64a9d49d3'
-        'SKIP'
         '0e9a6bbc9baf4e6706699257e811dcdb7d7e6c946a45f660ee56c564d907efaaac53387b29668ca3fc3082c5badc30ee082dac9d8de2bde72c79365af4050b47'
         'SKIP'
-        'bef3377ad86440af76e9dde4c29c9f4aaad42f5fe343f7d31f5eb537d6d358602f996f5d63986af275f2e92f94e71dc28c320edc8c03d05bd64dbd8ed23d75dc')
+        'SKIP'
+        'SKIP'
+        'SKIP')
 
 export KBUILD_BUILD_HOST=archlinux
 export KBUILD_BUILD_USER=$pkgbase
@@ -67,6 +72,11 @@ prepare() {
   echo "-$pkgrel" > localversion.10-pkgrel
   echo "${pkgbase#linux}" > localversion.20-pkgname
 
+  mkdir -p security/tlsm
+  mv ../tlsm-src/* security/tlsm/
+  patch -N security/Kconfig ../Kconfig.diff
+  patch -N security/Makefile ../Makefile.diff
+
   local src
   for src in "${source[@]}"; do
     src="${src%%::*}"
@@ -82,6 +92,8 @@ prepare() {
   make olddefconfig
   diff -u ../config .config || :
 
+  #make menuconfig
+
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
 }
@@ -90,7 +102,7 @@ build() {
   cd $_srcname
   make all
   make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
-  make htmldocs SPHINXOPTS=-QT
+  #make htmldocs SPHINXOPTS=-QT
 }
 
 _package() {
@@ -232,23 +245,23 @@ _package-docs() {
   cd $_srcname
   local builddir="$pkgdir/usr/lib/modules/$(<version)/build"
 
-  echo "Installing documentation..."
-  local src dst
-  while read -rd '' src; do
-    dst="${src#Documentation/}"
-    dst="$builddir/Documentation/${dst#output/}"
-    install -Dm644 "$src" "$dst"
-  done < <(find Documentation -name '.*' -prune -o ! -type d -print0)
-
-  echo "Adding symlink..."
-  mkdir -p "$pkgdir/usr/share/doc"
-  ln -sr "$builddir/Documentation" "$pkgdir/usr/share/doc/$pkgbase"
+  # echo "Installing documentation..."
+  # local src dst
+  # while read -rd '' src; do
+  #   dst="${src#Documentation/}"
+  #   dst="$builddir/Documentation/${dst#output/}"
+  #   install -Dm644 "$src" "$dst"
+  # done < <(find Documentation -name '.*' -prune -o ! -type d -print0)
+
+  # echo "Adding symlink..."
+  # mkdir -p "$pkgdir/usr/share/doc"
+  # ln -sr "$builddir/Documentation" "$pkgdir/usr/share/doc/$pkgbase"
 }
 
 pkgname=(
   "$pkgbase"
-  "$pkgbase-headers"
-  "$pkgbase-docs"
+  #"$pkgbase-headers"
+  #"$pkgbase-docs"
 )
 for _p in "${pkgname[@]}"; do
   eval "package_$_p() {
